drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/lockmanager/GuidedDecisionTableLockManagerImpl.java

```diff
@@ -49,7 +49,7 @@ public void fireChangeTitleEvent() {
             if (dtPath == null) {
                 return;
             }
-            if (dtPath.equals(path)) {
+            if (path.equals(dtPath)) {
```

Comment:
@manstis yeah, this change is tricky. Here we want to use the `.equals` method from the `PathImpl` (`path`), not from the `ObservablePath` (`dtPath`).

(I covered it with tests to avoid some regression :-) if we change the line 52 back to the `if (dtPath.equals(path))`, the `testFireChangeTitleEvent_LockInfoUpdateForActiveDecisionTable` breaks).

Reply:
@karreiro Sounds like an ```equals(..)``` implementation is broken somewhere :-(

https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)

"It is symmetric: for any non-null reference values x and y, x.equals(y) should return true if and only if y.equals(x) returns true."

I wonder whether it is better to fix the offending implementation?

Reply:
@manstis Definitely, the implementation of the [ObservablePathImpl#equals](https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/backend/vfs/impl/ObservablePathImpl.java#L436-L450) is not symmetric. Sometimes it uses the [o.original](https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/backend/vfs/impl/ObservablePathImpl.java#L446), sometime it uses the [o](https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/backend/vfs/impl/ObservablePathImpl.java#L449) ([here](https://github.com/kiegroup/appformer/blob/master/uberfire-api/src/main/java/org/uberfire/backend/vfs/PathFactory.java#L190-L202)).

My first approach was to change it. However, the current implementation is widely used and the instability that I created does worth. So, instead of that, I preferred to make a specific/small fix with less impact.

I definitely agree that `ObservablePathImpl#equals` needs to be refactored, but I think we could raise another JIRA for that.